### ISSUE TITLE
ComboBox selects a random option when clicking outside of the dropdown after scroll

### ISSUE THREAD

**Issue by c7i** [microsoft/fluentui#17008]
### Environment Information

- **Package version(s)**: Newest version 7.160.1 and also on 8.0.0-beta.55
- **Browser and OS versions**: Present on latest chrome (88.0.4324.150) and firefox

### Please provide a reproduction of the bug in a codepen:

https://codepen.io/frts/pen/OJbpGXE

1. Open dropdown
2. Scroll while hovering over the items (without hovering over the items it does not occur)
3. Immediately click outside the dropdown to dismiss it. Some random option is now selected.

![bug](https://user-images.githubusercontent.com/46483067/108055928-ebc23500-7050-11eb-8f29-9a00351fb881.gif)

#### Actual behavior:

Selects an item. Happens on both ComboBox and VirtualizedComboBox

#### Expected behavior:

The dropdown is dismissed by clicking outside of it, so nothing should be selected.

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: (Blocking, High, Normal, Low): Normal

**Comment by aidanmc95** [comment]
This is being created because of the interaction with the ', []' on line 34. The interaction between componentRef and [] causes this jumping. Just change line 34 with this, const onOpenClick = React.useCallback(() => comboBoxRef.current.focus(true));

**Comment by c7i** [comment]
> This is being created because of the interaction with the ', []' on line 34. The interaction between componentRef and [] causes this jumping. Just change line 34 with this, const onOpenClick = React.useCallback(() => comboBoxRef.current.focus(true));

I don't think it's the button or the ref. The only reason the button and the ComboBox ref is there is because I copied the codepen from the FluentUI components page. If you delete them, the issue is still present

**Comment by gouttierre** [comment]
@c7i - thanks for filling this issue. I did look at the repo (v7 and latest) and added the suggested code change. 
`line 34 `with  `const onOpenClick = React.useCallback(() => comboBoxRef.current.focus(true));` . The issue did not replicate as many times, but there we several instances the unusual behavior occurred on my end.

@khmakoto - Would you be able to confirm this one, if it is regression or if the behavior is an issue?

**Comment by xiaogangy** [comment]
Follow.
Is there any updates of this bug or is there any workaround to pass it?

**Comment by xiaogangy** [comment]
> > This is being created because of the interaction with the ', []' on line 34. The interaction between componentRef and [] causes this jumping. Just change line 34 with this, const onOpenClick = React.useCallback(() => comboBoxRef.current.focus(true));
> 
> I don't think it's the button or the ref. The only reason the button and the ComboBox ref is there is because I copied the codepen from the FluentUI components page. If you delete them, the issue is still present

Hi, do you have any idea how to resolve this problem? Thanks!

**Comment by MSpake** [comment]
We're encountering this in our app as well. Any solutions?
